babel/_compat.py

```diff
@@ -54,3 +56,74 @@
 
 
 number_types = integer_types + (float,)
+
+
+def _zip_split(path):
+    for ext in ('zip', 'whl'):
+        if '.' + ext + os.sep in path:
+            return ext
+    return None
+
+
+def extract_nested_zipfile(path, parent_zip=None):
```

Comment:
Can you add some unit tests for those new functions you're adding to the project?